parser/openfoodfacts_taxonomy_parser/parser.py

```diff
@@ -0,0 +1,336 @@
+                self.is_before = data["id"]
+                data = self.new_node_data()
+
+            # harvest the line
+            if not (line) or line[0] == "#":
+                data["preceding_lines"].append(line)
+                if not data[
+                    "src_position"
+                ]:  # to get the position of the footer if it's not empty
+                    data["src_position"] = line_number + 1
+            else:
+                if (
+                    len(data) == 5 and not data["parent_tag"]
+                ):  # the beginning of the entry
+                    data["src_position"] = line_number + 1
+                if "stopword" in line:
+                    id = "stopwords:" + str(index_stopwords)
+                    data = self.set_data_id(data, id, line_number)
+                    index_stopwords += 1
+                    lc, value = self.get_lc_value(line[10:])
+                    data["tags_" + lc] = value
+                    self.stopwords[lc] = value
+                elif "synonym" in line:
+                    id = "synonyms:" + str(index_synonyms)
+                    data = self.set_data_id(data, id, line_number)
+                    index_synonyms += 1
+                    line = line[9:]
+                    tags = [words.strip() for words in line[3:].split(",")]
+                    lc, value = self.get_lc_value(line)
+                    data["tags_" + lc] = tags
+                    data["tags_ids_" + lc] = value
+                elif line[0] == "<":
+                    data["parent_tag"].append(self.add_line(line[1:]))
+                elif language_code_prefix.match(line):
+                    if not data["id"]:
+                        data["id"] = self.add_line(line.split(",", 1)[0])
+                        data["main_language"] = data["id"][
+                            :2
+                        ]  # first 2 characters are language code
+                    # add tags and tagsid
+                    lang, line = line.split(":", 1)
+                    tags_list = []
+                    tagsids_list = []
+                    for word in line.split(","):
+                        tags_list.append(word.strip())
+                        word_normalized = self.remove_stopwords(
+                            lang, self.normalizing(word, lang)
+                        )
+                        tagsids_list.append(word_normalized)
+                    data["tags_" + lang] = tags_list
+                    data["tags_ids_" + lang] = tagsids_list
+                else:
+                    property_name, lc, property_value = line.split(":", 2)
+                    data["prop_" + property_name + "_" + lc] = property_value
+        data["id"] = "__footer__"
+        data["preceding_lines"].pop(0)
+        if not data["src_position"]:
+            data["src_position"] = line_number + 1  # to get position if it's empty
```

Comment:
as stated above here you can do:

```suggestion
        data["src_position"] = line_number - len(data["preceding_line"])
```